prometheus_hardware_exporter/collector.py

```diff
@@ -564,14 +567,43 @@ def _get_sensor_value_from_reading(self, reading: str) -> float:
             return 0.0
 
 
-class IpmiSelCollector(BlockingCollector):
+class IpmiSelCollector(NonBlockingCollector):
     """Collector for IPMI SEL data."""
 
     def __init__(self, config: Config) -> None:
         """Initialize the collector."""
         self.ipmi_sel = IpmiSel(config)
         super().__init__(config)
 
+        self._cache_timestamp = datetime.datetime.now().timestamp()
+        self._lock = threading.Lock()
+        self._update_thread = threading.Thread(target=self.update_cache, daemon=True)
+        self._update_thread.start()
+
+    def update_cache(self) -> None:
+        """Background thread function to update SEL cache periodically."""
+        logger.info("Starting IPMI SEL cache update thread.")
+
+        while True:
+            start_time = datetime.datetime.now()
+
+            sel_entries = self.ipmi_sel.get_sel_entries(self.config.ipmi_sel_interval)
+            with self._lock:
+                self._cache = sel_entries
+            self._cache_timestamp = datetime.datetime.now().timestamp()
+            logger.info("Updated IPMI SEL cache.")
```

Comment:
I have a concern that if the `get_sel_entries` fails for some reason the while True loop will stop executing because the thread dies andnNo further cache updates will occur, as the background thread is no longer running.

I think it's safer to add a try/except and log the error. Something like:

```python
try:
                sel_entries = self.ipmi_sel.get_sel_entries(self.config.ipmi_sel_interval)
                with self._lock:
                    self._cache = sel_entries
                self._cache_timestamp = datetime.datetime.now().timestamp()
                logger.info("Updated IPMI SEL cache.")
            except Exception as e:
                logger.error(f"Failed to update IPMI SEL cache: {str(e)}")
```